File: xmp/ZeroconfListener.js
#!/usr/bin/env node

var gepard = require ( "gepard" ) ;

// Option 1: callback
var client = gepard.getClient ( 'test-gepard', service => { return true ; } ) ;
/*
// Option 2: callback with decision
var client = gepard.getClient ( 'test-gepard', service => { if ( service.isLocalHost() ) return true ; } ) ;
// Option 3: auto-callback with timeout
var client = gepard.getClient ( { timeout:10000, type:'test-gepard' } ) ;
// Option 4: auto-callback type only
var client = gepard.getClient ( 'test-gepard' ) ;
// Option 5: auto-callback type given by: --gepard.zeroconf.type=test-gepard
var client = gepard.getClient() ;
*/

client.on ( "ALARM", (e) => console.log ( e ) ) ;
client.on ( "error", e => console.log ( e ) ) ;

File: python/xmp/ZeroconfListener.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )
import gepard
import time
# ==========================================================================

gepard.util.exitWithSIGINT()

def acceptService ( client, service ):
	print ( service )
	return True

def on_ALARM ( event ):
	print ( event )

gepard.Client.getInstance('test-gepard',acceptService).setReconnect ( True ).on ( "ALARM", on_ALARM )

